letsencrypt/client/auth_handler.py

```diff
@@ -387,13 +395,14 @@ def _find_dumb_path(challs, preferences):
     assert len(preferences) == len(set(preferences))
 
     path = []
-    satisfied = set()
+    # This cannot be a set() because POP challenge is not currently hashable
```

Comment:
Given that the set is only beneficial as an optimization if the server happens to send the exact same challenge multiple times (which we are already entering an undefined area here).  I will attempt to change it back after I go over #327.